### PR TITLE
fix: OPTIC-1420: Inner IDs are duplicated and counted wrong

### PR DESCRIPTION
**Pull request by mcanu**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [x] Backend (Database)
- [ ] Backend (API)
- [ ] Frontend

### Describe the reason for change
There was an issue where the inner_id was duplicated when importing from the API. We identified the problem because a client was importing with two requests almost at the same time, causing a race condition when calculating the inner_id.

#### What does this fix?
Potential duplicate inner ids due to a race condition when importing from the API.

#### What is the new behavior?
Acquire lock on the project before calculating inner id. I tried to acquire the lock on the Tasks queryset, but when there weren't any tasks yet it wouldn't work, because it locks on the returned rows (empty the first time).

#### Does this change affect performance?
If there are multiple async imports for a same project at once like this case, they will eventually be committed sequentially. This is needed to ensure inner_id consistency.

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit

For testing I created a small script to replicate this behavior. Running two separate worker processes is also needed, so they can consume each import job concurrently.

```python
import json
from label_studio_sdk.client import LabelStudio

# Define the URL where Label Studio is accessible and the API key for your user account
LABEL_STUDIO_URL = ''
# API key is available at the Account & Settings > Access Tokens page in Label Studio UI
API_KEY = ''


# Connect to the Label Studio API and check the connection
ls = LabelStudio(base_url=LABEL_STUDIO_URL, api_key=API_KEY)

project_id = 7

with open('data/5400.json', 'r') as f:
    data_5400 = json.load(f)

with open('data/5500.json', 'r') as f:
    data_5500 = json.load(f)


ls.projects.import_tasks(id=project_id, request=data_5400)
ls.projects.import_tasks(id=project_id, request=data_5500)
```

### Which logical domain(s) does this change affect?
Imports

### CODE REVIEW

**Comment by mcanu** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/12358429877)
> Successfully merged:  27 files changed, 842 insertions(+), 57 deletions(-)

**Comment by mcanu** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/12396610560)
> Successfully merged:  create mode 100644 web/libs/datamanager/src/components/MainView/GridView/ImagePreview.tsx

File: label_studio/tasks/serializers.py [hunk 1]
```diff
@@ -533,10 +533,13 @@ def add_tasks(self, task_annotations, task_predictions, validated_tasks):
         db_tasks = []
         max_overlap = self.project.maximum_annotations
 
-        # identify max inner id
-        tasks = Task.objects.filter(project=self.project)
-        prev_inner_id = tasks.order_by('-inner_id')[0].inner_id if tasks else 0
+        # Acquire a lock on the project to ensure atomicity when calculating inner_id
+        project = Project.objects.select_for_update().get(id=self.project.id)
+
+        last_task = Task.objects.filter(project=project).order_by('-inner_id').first()
```
Comment (makseq) [inline label_studio/tasks/serializers.py]: use fast_first instead, first() might be very slow

Comment (makseq) [reply label_studio/tasks/serializers.py]: or use [0] as it was before

**Review by makseq — CHANGES_REQUESTED** [review]

**Review by wesleylima — APPROVED** [review]

**Review by makseq — APPROVED** [review]

**Pull request merged**